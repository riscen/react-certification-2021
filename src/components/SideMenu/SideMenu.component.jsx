import React from 'react';
import CloseIcon from '@material-ui/icons/Close';

import {
  SideMenuContainer,
  MenuItem,
  CloseIconContainer,
  CloseIconSpan,
  MenuSeparator,
} from './SideMenu.modules';

function SideMenu(props) {
  const { toggleSideMenu } = props;

  return (
    <SideMenuContainer>
      <CloseIconContainer onClick={() => toggleSideMenu(false)}>
        <CloseIconSpan>
          <CloseIcon />
        </CloseIconSpan>
      </CloseIconContainer>
      <MenuSeparator />
      <MenuItem>
        <span>This is the side menu :)</span>
      </MenuItem>
    </SideMenuContainer>
  );
}

export default SideMenu;
