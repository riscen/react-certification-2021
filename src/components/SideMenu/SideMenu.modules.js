import styled from 'styled-components';

export const SideMenuContainer = styled.div`
  background-color: #c2c3ff;
  color: black;
  height: 100%;
  overflow: hidden;
`;

export const MenuItem = styled.div`
  margin: 20px;
`;

export const CloseIconContainer = styled.div`
  display: flex;
  height: 62px !important;
  justify-content: flex-end;
  margin-right: 10px;
  padding: 20px;
`;

export const CloseIconSpan = styled.span`
  cursor: pointer;
`;

export const MenuSeparator = styled.div`
  border: 1px solid #ffffff;
`;
