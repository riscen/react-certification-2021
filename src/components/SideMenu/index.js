export { default } from './SideMenu.component';
export {
  SideMenuContainer,
  MenuItem,
  CloseIconContainer,
  CloseIconSpan,
  MenuSeparator,
} from './SideMenu.modules';
