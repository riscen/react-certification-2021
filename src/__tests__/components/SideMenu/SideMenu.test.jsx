import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { mount, configure } from 'enzyme';
import CloseIcon from '@material-ui/icons/Close';

import SideMenu, {
  SideMenuContainer,
  MenuItem,
  CloseIconContainer,
  CloseIconSpan,
  MenuSeparator,
} from '../../../components/SideMenu';

describe('SideMenu', () => {
  beforeEach(() => {
    configure({ adapter: new Adapter() });
  });

  it('Renders SideMenu', () => {
    const wrapper = mount(<SideMenu toggleSideMenu={jest.fn()} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('Renders SideMenuContainer', () => {
    const wrapper = mount(<SideMenu toggleSideMenu={jest.fn()} />);
    expect(wrapper.find(SideMenuContainer)).toBeDefined();
    wrapper.unmount();
  });

  it('Renders CloseIconContainer', () => {
    const wrapper = mount(<SideMenu toggleSideMenu={jest.fn()} />);
    expect(wrapper.find(CloseIconContainer)).toBeDefined();
    wrapper.unmount();
  });

  it('Renders CloseIconSpan', () => {
    const wrapper = mount(<SideMenu toggleSideMenu={jest.fn()} />);
    expect(wrapper.find(CloseIconSpan)).toBeDefined();
    wrapper.unmount();
  });

  it('Renders CloseIcon', () => {
    const wrapper = mount(<SideMenu toggleSideMenu={jest.fn()} />);
    expect(wrapper.find(CloseIcon)).toBeDefined();
    wrapper.unmount();
  });

  it('Renders MenuSeparator', () => {
    const wrapper = mount(<SideMenu toggleSideMenu={jest.fn()} />);
    expect(wrapper.find(MenuSeparator)).toBeDefined();
    wrapper.unmount();
  });

  it('Displays side menu content', () => {
    const textContent = 'This is the side menu :)';
    const wrapper = mount(<SideMenu toggleSideMenu={jest.fn()} />);
    expect(wrapper.find(MenuItem).find('span').text()).toEqual(textContent);
    wrapper.unmount();
  });
});
