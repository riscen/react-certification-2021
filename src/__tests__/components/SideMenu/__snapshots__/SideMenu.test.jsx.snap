// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideMenu Renders SideMenu 1`] = `ReactWrapper {}`;
